### PR TITLE
Prepare for 4.1.0

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
# Description

I'm proposing this as the code that'll become 4.1.0, when it's approved+merged.

It's possible the release isn't done on 3 March 2024 as the CHANGELOG proposes, but there shouldn't be changes to the main branch after this pull request either, so that should be Ok.

- [x] I have performed a self-review of my changes
- [x] I have read and understood the [contributing guidelines](/kerl/kerl/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+  - Fix/complement `cleanup` (#492)
+  - Be more specific in the packages we suggest for installation (#491)
+  - Fix broken Zsh completion scripts (#497)
+  - `brew` in CI: don't install updates "just because" (#499)
+  - Attempt auto-cleanup of `otp_builds` and `otp_installations`, at every run (#490)
+  - Make Pop!_OS a `kerl`-known Linux distro, for package probing (#505)
+  - Prevent GitHub CI warnings on "Node.js 16 actions are deprecated" (#506)
+  - Fix shell completion scripts (#501)
+  - Extend `KERL_AUTOCLEAN` to Git-based builds (#511)
+
```
Comment (paulo-ferraz-oliveira) [inline CHANGELOG.md:27]:
```suggestion
  - Extend `KERL_AUTOCLEAN` to Git-based builds (#511)
  - Have CI run on pull request, workflow dispatch and push to main (#516)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,30 @@
...
+  - Fix/complement `cleanup` (#492)
+  - Be more specific in the packages we suggest for installation (#491)
+  - Fix broken Zsh completion scripts (#497)
+  - `brew` in CI: don't install updates "just because" (#499)
+  - Attempt auto-cleanup of `otp_builds` and `otp_installations`, at every run (#490)
+  - Make Pop!_OS a `kerl`-known Linux distro, for package probing (#505)
+  - Prevent GitHub CI warnings on "Node.js 16 actions are deprecated" (#506)
+  - Fix shell completion scripts (#501)
+  - Extend `KERL_AUTOCLEAN` to Git-based builds (#511)
+  - Have CI run on pull request, workflow dispatch and push to main (#516)
```
Comment (paulo-ferraz-oliveira) [inline CHANGELOG.md:27]:
```suggestion
  - Have CI run on pull request, workflow dispatch and push to main (#516)
  - Ignore ownership and permissions of files in OTP source archives (#515)
  
```

**Comment by paulo-ferraz-oliveira** [comment]
> there shouldn't be changes to the main branch after this pull request either

There were, but I merged them into the CHANGELOG, too. 😄 

@jadeallenx, care to give it a look-see? Thanks.

**Review by jadeallenx — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
... aaaand released!

**Pull request merged**